datafusion/functions/src/core/getfield.rs

```diff
@@ -184,9 +184,29 @@ impl ScalarUDFImpl for GetFieldFunc {
         };
 
         match (array.data_type(), name) {
-            (DataType::Map(_, _), ScalarValue::Utf8(Some(k))) => {
+            (DataType::Map(_, _), name) => {
                 let map_array = as_map_array(array.as_ref())?;
-                let key_scalar: Scalar<arrow::array::GenericByteArray<arrow::datatypes::GenericStringType<i32>>> = Scalar::new(StringArray::from(vec![k.clone()]));
+                if !matches!(name, ScalarValue::Utf8(_) | ScalarValue::Int64(_) | ScalarValue::Float64(_)) {
+                    return exec_err!(
+                "get indexed field is only possible on map with utf8, int64 and float64 indexes. \
+                             Tried with {name:?} index")
+                }
+                let key_array: ArrayRef = match name {
+                    ScalarValue::Int64(Some(k)) => {
+                        Arc::new(Int64Array::from(vec![*k]))
+                    }
+                    ScalarValue::Utf8(Some(k)) => {
+                        Arc::new(StringArray::from(vec![k.clone()]))
+                    }
+                    ScalarValue::Float64(Some(k)) => {
+                        Arc::new(Float64Array::from(vec![*k]))
+                    }
+                    _ => {
+                        unreachable!();
```

Comment:
Instead of `unreachable` panic, I prefer to throw the Datafuion error `internal_err` like
```
return internal_err!("Unexpected scalar value type: {name}");
```
Although I think L189 has protected it, if any change breaks this protection, we can find it easily.